src/Dialog/Dialog.js

```diff
@@ -105,6 +107,8 @@ function getStyles(props, context) {
       transition: open ?
         transitions.easeOut('0ms', 'left', '0ms') :
         transitions.easeOut('0ms', 'left', '450ms'),
+      direction: isRtl ? 'rtl' : 'ltr',
```

Comment:
I *personally* favor the first approach for the following reasons:
- Depending on how components are nested, the direction may not be always passed down to components at deeper level properly. 
- I assume users have already set muiTheme.isRtl to true in these right-to-left rendering scenarios. Asking them to ALSO specify direction at 'body' element seems redundant and confusing.